system-int-test/test/cmr/system_int_test/ingest/deleted_granules_index_test.clj

```diff
@@ -0,0 +1,56 @@
+(ns cmr.system-int-test.ingest.deleted-granules-index-test
+  "When a granule is deleted, a document is indexed in the deleted-granules index.
+   This namespace tests that functionality"
+  (:require
+   [clojure.test :refer :all]
+   [cmr.metadata-db.services.concept-service :as concept-service]
+   [cmr.system-int-test.data2.core :as d]
+   [cmr.system-int-test.data2.granule :as dg]
+   [cmr.system-int-test.data2.umm-spec-collection :as data-umm-c]
+   [cmr.system-int-test.utils.dev-system-util :as dev-sys-util]
+   [cmr.system-int-test.utils.index-util :as index]
+   [cmr.system-int-test.utils.ingest-util :as ingest]
+   [cmr.system-int-test.utils.metadata-db-util :as mdb-util]))
+
+(use-fixtures :each (ingest/reset-fixture {"provguid1" "PROV1"}))
+
+(defn check-index-for-deleted-granule
+  "Check elastic search deleted-granules index from related deleted granule entry,
+   Returns true if document exists, false if it does not."
+  [concept-id]
+  (index/doc-present? "deleted-granules" "deleted-granule" concept-id))
+
+(deftest deleted-granules-test
+  (testing "Ingest granule, delete, then reingest"
+    (let [collection (d/ingest-umm-spec-collection "PROV1" (data-umm-c/collection {}))
+          granule (d/item->concept (dg/granule-with-umm-spec-collection collection (:concept-id collection) {:concept-id "G1-PROV1"}))
+          ingest-result (ingest/ingest-concept granule)
+          delete-result (ingest/delete-concept granule)
+          ingest-revision-id (:revision-id ingest-result)
+          delete-revision-id (:revision-id delete-result)]
+      (index/wait-until-indexed)
+      (Thread/sleep 1000)
```

Comment:
Why do we need to sleep?

`wait-until-indexed` should block until all messages on the message queue have been processed. I'd prefer to remove the sleeps because they could be masking some problem or inconsistency with the message queue processing.

Reply:
I was testing for a timing problem with a failing build in CI, only temporary.